frontend/public/components/storage/attach-storage.tsx

```diff
@@ -40,113 +27,136 @@ const PVCDropdown: React.SFC<PVCDropdownProps> = props => {
-  > {
-  state = {
-    resourceObj: null,
-    inProgress: false,
-    claimName: '',
-    volumeName: '',
-    mountPath: '',
-    subPath: '',
-    mountAsReadOnly: false,
-    allContainers: true,
-    containers: {},
-    volumeAlreadyMounted: false,
-    error: '',
-    showCreatePVC: 'existing',
-    newPVCObj: null,
-  };
+export const AttachStorageForm: React.FC<AttachStorageFormProps> = (props) => {
+  const [resourceObj, setResourceObj] = React.useState(null);
+  const [inProgress, setInProgress] = React.useState(false);
+  const [useContainerSelector, setUseContainerSelector] = React.useState(false);
+  const [claimName, setClaimName] = React.useState('');
+  const [volumeName, setVolumeName] = React.useState('');
+  const [mountPath, setMountPath] = React.useState('');
+  const [subPath, setSubPath] = React.useState('');
+  const [mountAsReadOnly, setMountAsReadOnly] = React.useState(false);
+  const [allContainers, setAllContainers] = React.useState(true);
+  const [selectedContainers, setSelectedContainers] = React.useState([]);
+  const [volumeAlreadyMounted, setVolumeAlreadyMounted] = React.useState(false);
+  const [error, setError] = React.useState('');
+  const [showCreatePVC, setShowCreatePVC] = React.useState('existing');
+  const [newPVCObj, setNewPVCObj] = React.useState(null);
 
-  componentDidMount() {
-    const { kindObj, name, namespace } = this.props;
-    const supportedKinds = [
-      'Deployment',
-      'DeploymentConfig',
-      'ReplicaSet',
-      'ReplicationController',
-      'StatefulSet',
-      'DaemonSet',
-    ];
-
-    if (!kindObj || !_.includes(supportedKinds, kindObj.kind)) {
-      this.setState({ error: 'Unsupported kind.' });
-      return;
-    }
+  const { kindObj, resourceName, namespace } = props;
+  const supportedKinds = [
+    'Deployment',
+    'DeploymentConfig',
+    'ReplicaSet',
+    'ReplicationController',
+    'StatefulSet',
+    'DaemonSet',
+  ];
 
+  React.useEffect(() => {
     // Get the current resource so we can add to its definition
-    k8sGet(kindObj, name, namespace).then(
-      resourceObj => {
-        this.setState({ resourceObj });
+    k8sGet(kindObj, resourceName, namespace).then(
+      retrievedResourceObj => {
+        setResourceObj(retrievedResourceObj);
       },
-      err => this.setState({ error: err.message })
+      err => setError(err.message)
     );
+  }, [kindObj, resourceName, namespace]);
+
+  if (!kindObj || !_.includes(supportedKinds, kindObj.kind)) {
+    setError('Unsupported kind.');
+    return;
   }
 
-  updateVolumeName = (claimName: string) => {
-    const { resourceObj } = this.state;
+  const updateVolumeName = (newClaimName: string) => {
     // Check if there is already a volume for this PVC.
     const volumes = _.get(resourceObj, 'spec.template.spec.volumes');
     const volume = _.find(volumes, {
       persistentVolumeClaim: {
-        claimName,
+        claimName: newClaimName,
       },
     }) as any;
```

Comment:
```suggestion
    }) as Volume;
```

Reply:
The Volume type expects a `name` field instead of `claimName`so this change does not work.